src/Microsoft.Graph/Serialization/DateConverter.cs

```diff
@@ -0,0 +1,68 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph
+{
+    using System;
+
+    using Newtonsoft.Json;
+
+    public class DateConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            if (objectType == typeof(Date))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            try
+            {
+                var dateTime = (DateTime)serializer.Deserialize(reader, typeof(DateTime));
+
+                return new Date(dateTime);
+            }
+            catch (JsonSerializationException serializationException)
+            {
+                throw new ServiceException(
+                    new Error
+                    {
+                        Code = GraphErrorCode.GeneralException.ToString(),
+                        Message = string.Format("Unable to deserialize the returned Date."),
+                    },
+                    serializationException);
+            }
+        }
+
+        /// <summary>
+        /// Writes the JSON representation of the object.
+        /// </summary>
+        /// <param name="writer">The <see cref="JsonWriter"/> to write to.</param>
+        /// <param name="value">The value.</param>
+        /// <param name="serializer">The calling serializer.</param>
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var date = value as Date;
+
+            if (date != null)
+            {
+                writer.WriteValue(date.ToString());
+            }
+            else
+            {
+                throw new ServiceException(
+                    new Error
+                    {
+                        Code = GraphErrorCode.GeneralException.ToString(),
+                        Message = string.Format("DateConverter can only serialize objects of type Date."),
```

Comment:
This error message seems more likely to be hit if the value is null to begin with, as opposed to being of the wrong type?